Clean up on the List and List-Method Isles (#2404)
exercism/python: exercises/concept/elyses-enchantments/.docs/introduction.md

# Introduction

In Python, a [list][list] is a mutable collection of items in _sequence_. Like most collections (_see the built-ins [`tuple`][tuple], [`dict`][dict] and [`set`][set]_), lists can hold reference to any (or multiple) data type(s) - including other lists.

> Like any [sequence][sequence type], items are referenced by 0-based index number, and can be copied in whole or in part via _slice notation_. Lists support all [common sequence operations][common sequence operations], as well as [mutable sequence operations][mutable sequence operations] such as `.append()` and `.reverse()`. They can be iterated over in a loop by using the `for item in <list>` construct.

Reviewer: ```suggestion
Like any [sequence][sequence type], items are referenced by 0-based index number, and can be copied in whole or in part via _slice notation_. `list`s support all [common sequence operations][common sequence operations], as well as [mutable sequence operations][mutable sequence operations] such as `<list>.append()` and `<list>.reverse()`. They can be iterated over in a loop by using the `for item in <list>` construct.
```

## Accessing elements

> Items inside lists (_as well as items in other sequence types `string` & `tuple`_), can be accessed via 0-based index and _bracket notation_. Indexes can be from **`left`** --> **`right`** (_starting at zero_) or **`right`** --> **`left`** (_starting at -1_).

Reviewer: ```suggestion
Items inside lists (_as well as items in other sequence types `string` & `tuple`_) can be accessed via 0-based index and _bracket notation_. Indexes can be from **`left`** --> **`right`** (_starting at zero_) or **`right`** --> **`left`** (_starting at -1_).
```

>>> breakfast_foods[-4]
'Oatmeal'

# Eggs are at index -2 or 2
>>> breakfast_foods[-2]
'Eggs'

>>> breakfast_foods[2]
'Eggs'

# Toast is at -1
>>> breakfast_foods[-1]
'Toast'
```

> A portion of a list can be accessed via _slice notation_ (`<list>[start:stop]`). A _slice_ is defined as an element sequence at position `index`, such that `start <= index < stop`. _Slicing_ returns a copy of the "sliced" items and does not modify the original `list`.

Reviewer: ```suggestion
A section of the elements inside a list can be accessed via _slice notation_ (`<list>[start:stop]`). A _slice_ is defined as an element sequence at position `index`, such that `start <= index < stop`. _Slicing_ returns a copy of the "sliced" items and does not modify the original `list`.
```
